skyportal/handlers/api/internal/image_analysis.py

```diff
@@ -0,0 +1,607 @@
+from baselayer.app.access import permissions
+from baselayer.app.env import load_env
+from baselayer.log import make_log
+import os
+import arrow
+from astropy.io import fits
+from astropy.wcs import WCS
+from astropy.table import vstack
+from astropy.coordinates import SkyCoord
+from astropy.coordinates import search_around_sky
+from astropy.time import Time
+import astropy.units as u
+
+import base64
+import numpy as np
+import sqlalchemy as sa
+import tempfile
+from tqdm.auto import tqdm
+from tornado.ioloop import IOLoop
+import shutil
+
+from ..photometry import add_external_photometry
+from ...base import BaseHandler
+from ....models import Instrument, User, DBSession
+
+_, cfg = load_env()
+
+log = make_log('image_analysis')
+
+try:
+    if cfg['image_analysis'] is True:
+        import astroscrappy
+        from stdpipe import (
+            astrometry,
+            photometry,
+            catalogs,
+            cutouts,
+            templates,
+            plots,
+            pipeline,
+            psf,
+            utils,
+        )
+except Exception as e:
+    log(e)
+
+try:
+    if cfg['image_analysis'] is True:
+        # remove any temp dir starting with 'sex'
+        for dir in os.listdir('/tmp'):
+            if dir.startswith('sex') or dir.startswith('psfex'):
+                shutil.rmtree(os.path.join('/tmp', dir))
+except Exception as e:
+    log(e)
+
+catalogs_enum = [
+    "ps1",
+    "gaiadr2",
+    "gaiaedr3",
+    "skymapper",
+    "vsx",
+    "apass",
+    "sdss",
+    "atlas",
+    "usnob1",
+    "gsc",
+]
+
+templates_enum = ["PanSTARRS/DR1/g", "PanSTARRS/DR1/r", "PanSTARRS/DR1/i"]
+
+methods_enum = ["scamp", "astropy", "astrometrynet"]
+
+
+def spherical_match(ra1, dec1, ra2, dec2, sr=1 / 3600):
```

Comment:
One of the things we could consider putting as a tunable parameter (either in the config or as a user input or both) is the radius for the cross match. This is notoriously hard to pin down to one universal value. It depends on the field density and the survey resolution, among other considerations.

Reply:
Agreed. I will for sure add that in the next PR, hard-coding it is not optimal at all. However I guess we need to put some sort of constraints to avoid values that are too large.